### DISCUSSION TITLE
ui- classes outside HeadlessUI

### DISCUSSION THREAD

**Discussion by inwardmovement** [tailwindlabs/headlessui#1619] [Ideas]
Are there plans to make [`@headlessui-tailwindcss`](https://github.com/tailwindlabs/headlessui/tree/main/packages/%40headlessui-tailwindcss) package's `ui-` classes available outside HeadlessUI? Vanilla JS, or AlpineJS...

**Comment by RobinMalfait** [comment]
You can technically already use them outside of Headless UI if you want. The only requirement is that you provide `data-headlessui-state="..."` attribute to the elements. Not sure if we are going to make that part configurable, maybe if there is more demand for it.

**Reply by inwardmovement** [reply]
Nice!
